expr/literals.go

```diff
@@ -408,19 +465,85 @@ type ProtoLiteral struct {
 
 func (t *ProtoLiteral) ValueString() string {
 	switch literalType := t.Type.(type) {
-	case *types.PrecisionTimestampType, *types.PrecisionTimestampTzType:
-		return fmt.Sprintf("%d", t.Value)
+	case *types.PrecisionTimestampType:
+		x, _ := t.Value.(int64)
+		var tm time.Time
+		switch literalType.Precision {
+		case types.PrecisionSeconds:
+			tm = time.Unix(x, 0)
+		case types.PrecisionDeciSeconds:
+			tm = time.Unix(x/10, x%10*100000000)
+		case types.PrecisionCentiSeconds:
+			tm = time.Unix(x/100, x%100*10000000)
+		case types.PrecisionMilliSeconds:
+			tm = time.UnixMilli(x)
+		case types.PrecisionEMinus4Seconds:
+			tm = time.Unix(x/10000, x%10000*100000)
+		case types.PrecisionEMinus5Seconds:
+			tm = time.Unix(x/100000, x%100000*10000)
+		case types.PrecisionMicroSeconds:
+			tm = time.UnixMicro(x)
+		case types.PrecisionEMinus7Seconds:
+			tm = time.Unix(x/10000000, x%10000000*100)
+		case types.PrecisionEMinus8Seconds:
+			tm = time.Unix(x/100000000, x%100000000*10)
+		case types.PrecisionNanoSeconds:
+			tm = time.Unix(x/1000000000, x%1000000000)
+		default:
+			panic("unsupported precision")
+		}
+		return tm.UTC().Format("2006-01-02 15:04:05.999999999")
```

Comment:
this and the part below it are identical, can we push this to a helper function and remove the duplication?

Reply:
done